pkg/component/runtime/service.go

```diff
@@ -226,7 +225,7 @@ func (s *serviceRuntime) Run(ctx context.Context, comm Communicator) (err error)
 				s.forceCompState(client.UnitStateFailed, err.Error())
 			}
 		case newComp := <-s.compCh:
-			s.processNewComp(newComp, comm)
+			s.processNewComp(newComp, comm, true)
```

Comment:
Why do you sometimes need to send the observed state and sometimes not? Can this be clarified with comments?

Reply:
With the introduction of the tamper protection there was a need for another internal step for the service runtime in order to handle the Teardown sequence correctly for the protected Endpoint.
For the case of unprotected Endpoint the Teardown immediately led to onStop and only one state change was sent back.

For the case of protected Endpoint the ```Teardown``` sequence sends another configuration update to Endpoint in order to unprotect the Endpoint before it can be stopped/uninstalled.
If service runtime sends two configuration updates ```onTeardown``` from the service runtime the second one gets blocked on attempting to write to the channel:
https://github.com/elastic/elastic-agent/blob/main/pkg/component/runtime/service.go#L524

The solution here was to make the state change consistent/same for both protected and unprotected cases:
* Agent receives the consistent state update onTeardown in both cases (it doesn't really care about additional step for the protected Endpoint teardown before it can be stopped).
* The internals of the "teardown" sequence are encapsulated within the service runtime implementation.